### PR TITLE
In MVD3, don't read the whole library dataset for 1 element ranges.

### PR DESCRIPTION
**Pull request by hernando**
The function resolve_index reads the whole library dataset regardless of the necessary indices. In circuits with unique morphologies, this is extremely wasteful when client code wants to resolve the morphology names of just a few indices. Since the public API does not specify that getMorphologies, getMTypes, etc have linear complexity on the library size, client code can easily fall in this performance pitfall. This PR provides a fast path for single cell look ups.

Another client side solution is to use listAllMorphologies and getIndexMorphologies (which should be called getMorpholoyIndices in my opinion), however this requires the client to cache the morphology name list, which could be rather big. Ideally, the file format should not use a library dataset for morphology names when they are unique and MVDTool should detect which is the case at circuit opening. Then, getMorphologies would work the same as getPositions when the circuit has unique morphologies.

### CODE REVIEW

**Comment by hernando** [comment]
@matz-e, can C++11 be used in this project?

File: include/mvd/bits/mvd3_misc.hpp [hunk 1]
```diff
@@ -59,6 +59,15 @@ inline std::vector<T> get_data_for_selection(HighFive::DataSet & dataset, const
 
 template<typename T>
 inline std::vector<T> resolve_index(HighFive::DataSet & index, const MVD3::Range & range, HighFive::DataSet & data){
+    if (range.count == 1)
```
Comment (matz-e) [inline include/mvd/bits/mvd3_misc.hpp]:
```suggestion
    if (range.count == 1) {
```

File: include/mvd/bits/mvd3_misc.hpp [hunk 2]
```diff
@@ -59,6 +59,15 @@ inline std::vector<T> get_data_for_selection(HighFive::DataSet & dataset, const
...
+    {
```
Comment (matz-e) [inline include/mvd/bits/mvd3_misc.hpp]:
```suggestion
```

**Comment by hernando** [comment]
I reimplemented the solution to be more general. The code is not much more complicated.

**Pull request merged**